test/e2e/mcd_test.go

```diff
@@ -136,28 +157,16 @@ func TestMCDeployed(t *testing.T) {
 		if err != nil {
 			t.Errorf("%v", err)
 		}
-
-		visited := make(map[string]bool)
-		if err := wait.Poll(2*time.Second, 30*time.Minute, func() (bool, error) {
-			nodes, err := getNodesByRole(cs, "worker")
-			if err != nil {
-				return false, nil
-			}
-			for _, node := range nodes {
-				if visited[node.Name] {
-					continue
-				}
-				if node.Annotations[constants.CurrentMachineConfigAnnotationKey] == newMCName && node.Annotations[constants.MachineConfigDaemonStateAnnotationKey] == constants.MachineConfigDaemonStateDone {
-					visited[node.Name] = true
-					if len(visited) == len(nodes) {
-						return true, nil
-					}
-					continue
-				}
-			}
-			return false, nil
-		}); err != nil {
-			t.Errorf("machine config didn't result in file being on any worker: %v", err)
+		if err := waitForPoolComplete(cs, "worker", newMCName); err != nil {
```

Comment:
isn't thi racy if we jump here before the pool starts rolling? I believe that's why I went node by node confirming current is there and of the new pool

Reply:
ok, you added a check below but this may flake then?

Reply:
Just above that we do `waitForRenderedConfig` right?

Reply:
For the record though I didn't test this much locally, threw it up to have CI do it.

Reply:
oh yeah, I had an old master on my laptop and didn't notice that, cool yeah, no flake no race :tada: